.github/workflows/e2e.yml

```diff
@@ -0,0 +1,188 @@
+
+    - uses: actions/upload-artifact@v3
+      with:
+        name: test-image
+        path: _output/clustercidr-controller-image.tar
+
+  e2e:
+    name: e2e
+    runs-on: ubuntu-latest
+    timeout-minutes: 100
+    strategy:
+      fail-fast: false
+      matrix:
+        ipFamily: ["ipv4", "ipv6", "dual"]
+    env:
+      JOB_NAME: "clustercidr-e2e-${{ matrix.ipFamily }}"
+      IP_FAMILY: ${{ matrix.ipFamily }}
+    needs:
+      - build-image
+    steps:
+    - name: Check out code
+      uses: actions/checkout@v4
+
+    - name: Enable ipv4 and ipv6 forwarding
+      run: |
+        sudo sysctl -w net.ipv6.conf.all.forwarding=1
+        sudo sysctl -w net.ipv4.ip_forward=1
+
+    - name: Set up environment
+      run: ./hack/github/setup-test-env.sh ${{ env.K8S_VERSION }} ${{ env.KIND_VERSION }}
+
+    - name: Create multi node cluster
+      run: |
+        # output_dir
+        mkdir -p _artifacts
+        # create cluster
+        cat <<EOF | /usr/local/bin/kind create cluster \
+          --name ${{ env.KIND_CLUSTER_NAME}}           \
+          --image kindest/node:${{ env.K8S_VERSION }}  \
+          -v7 --wait 1m --retain --config=-
+        kind: Cluster
+        apiVersion: kind.x-k8s.io/v1alpha4
+        networking:
+          ipFamily: ${IP_FAMILY}
+        nodes:
+        - role: control-plane
+          kubeadmConfigPatches:
+          - |
+            kind: ClusterConfiguration
+            controllerManager:
+              extraArgs:
+                allocate-node-cidrs: "false"
+        - role: worker
+        - role: worker
+        EOF
+        /usr/local/bin/kind get kubeconfig --name ${{ env.KIND_CLUSTER_NAME}} > _artifacts/kubeconfig.conf
+
+    - uses: actions/download-artifact@v3
+      with:
+        name: test-image
+
+    - name: Install cluster-cidr-controller
+      run: |
+        # preload image
+        docker load --input clustercidr-controller-image.tar
+        /usr/local/bin/kind load docker-image ghcr.io/mneverov/cluster-cidr-controller:ci-e2e-test --name ${{ env.KIND_CLUSTER_NAME}}
+
+        # install CRD separately
+        /usr/local/bin/kubectl create -f charts/cluster-cidr-controller/gen/crds/networking.x-k8s.io_clustercidrs.yaml
+        
+        # create ClusterCIDR
+        /usr/local/bin/kubectl create -f samples/clustercidr.yaml
+        
+        # configure controller version and disable installing CRD
+        sed -i 's/tag.*/tag: "ci-e2e-test"/' charts/cluster-cidr-controller/values.yaml
+        sed -i 's/installCRDs: true/installCRDs: false/' charts/cluster-cidr-controller/values.yaml
+        
+        # install the controller
+        helm template charts/cluster-cidr-controller --values charts/cluster-cidr-controller/values.yaml  > temp.yaml
+        cat temp.yaml
+        /usr/local/bin/kubectl apply -f ./temp.yaml
+
+    - name: Get Cluster status
+      run: |
+        # wait network is ready
```

Comment:
ok, do this 
```suggestion
        # restart kindnet because of bug https://github.com/kubernetes-sigs/kind/pull/3438
         kubectl -n kube-system rollout restart ds kindnet
        # wait network is ready
```